### PR TITLE
Add mac

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
initial mac support 🎉

### CODE REVIEW

File: src/PoshNotify.psd1 [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+GUID = '742f6d08-546a-4598-8155-fab499b3c1c0'
+
+# Author of this module
+Author = 'idk'
+
+# Company or vendor of this module
+CompanyName = 'idk'
+
+# Copyright statement for this module
+Copyright = '(c) idk. All rights reserved.'
```
[the remark this answers is not in the document]
Comment (Windos) [reply src/PoshNotify.psd1]: Yeah, we can figure something out, maybe just all the contributors names? 😛

File: src/Public/Send-OSNotification.ps1 [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        $Icon
+    )
+
+    switch ($true) {
+        $IsWindows {
+            break
+        }
+        $IsMacOS {
+            if ($Icon) {
+                Write-Warning 'MacOS does not support the Icon parameter at this time.'
```
[the remark this answers is not in the document]
Comment (markwragg) [reply src/Public/Send-OSNotification.ps1]: Support for -Icon you mean? It supports images already but possibly under a different parameter name (can’t check right now).

[the remark this answers is not in the document]
Comment (Windos) [reply src/Public/Send-OSNotification.ps1]: It's tucked away in the "alerter notification" function, as "AppIcon": https://github.com/markwragg/PowerShell-MacNotify/blob/master/MacNotify/Public/Invoke-AlerterNotification.ps1

[the remark this answers is not in the document]
Comment (TylerLeonhardt) [reply src/Public/Send-OSNotification.ps1]: Oooo it's a different cmdlet!

**Comment by TylerLeonhardt** [comment]
Ah cool 👍 thanks for the info. Looks like the alerter one has what we need.

So `alerter` is some macOS util executable it seems and `Invoke-MacNotification` uses AppleScript?

**Comment by TylerLeonhardt** [comment]
I see. It looks like Alterter is also blocking - can't just fire off a notification without caring about the response.

I could start a Job but that's a lot of overhead for a notification.

I guess what I'll do is - if Icon is specified, I'll use
```powershell
Invoke-AlerterNotification -Title $Title -Message $Body -AppIcon $Icon -Timeout 2 -Silent
```

if not, I'll stick to what's in this PR.

**Pull request merged**